src/channel.rs

```diff
@@ -0,0 +1,84 @@
+use serde::{Deserialize, Serialize};
+
+use std::sync::{Arc, Mutex};
+use tonic_openssl_lnd::lnrpc::{self, channel_point};
+
+use crate::{AppState, MAX_SEND_AMOUNT};
+
+#[derive(Clone, Deserialize)]
+pub struct ChannelRequest {
+    capacity: i64,
+    push_amount: i64,
+    pubkey: String,
+    host: String,
```

Comment:
Should make this optional so the user can be connected already, that way we can skip the connect phase if needed

Reply:
Changed to `Option` and moved the connect into a conditional. I think this might be a _bit_ of a footgun as I imagine most people will be setting their node up for the first time for the faucet, but the UI can continue to suggest using the full host string.